presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -669,4 +671,62 @@ public static Boolean castTokenAsBoolean(JsonParser parser)
                 throw new PrestoException(INVALID_CAST_ARGUMENT, format("Unexpected token when cast to %s: %s", StandardTypes.BOOLEAN, parser.getText()));
         }
     }
+
+    public static Long castTokenAsShortDecimal(JsonParser parser, int precision, int scale)
+            throws IOException
+    {
+        BigDecimal bigDecimal = castTokenAsJavaDecimal(parser, precision, scale);
+        if (bigDecimal != null && bigDecimal.precision() > precision) {
+            // TODO: Should we use NUMERIC_VALUE_OUT_OF_RANGE instead?
+            throw new PrestoException(INVALID_CAST_ARGUMENT, format("Cannot cast input json to DECIMAL(%s,%s)", precision, scale));
```

Comment:
Use `parser.getText()` instead `"input json"` in error message.

Edit:

I see that it actually overridden by new `PrestoException` in `DecimalCasts.jsonToShortDecimal`.
I am not a big fan of that. I would rather see the final method `DecimalCasts.jsonToShortDecimal` construct `PrestoException` which is supposed to be user visible. And helper methods `JsonUtil` use internal exceptions like `NumberFormatException`. Or if helper method throws `PrestoException` it should be just rethrown by `DecimalCasts.jsonToShortDecimal`, not wrapped around.